deepchem/data/data_loader.py

```diff
@@ -415,29 +384,119 @@ def _get_shards(self, input_files, shard_size):
       List of filenames to process
     shard_size: int
       The size of a shard of data to process at a time.
+
+    Returns
+    -------
+    Iterator over shards
     """
     return load_csv_files(input_files, shard_size)
 
-  def _featurize_shard(self, shard):
-    """Featurizes a shard of an input dataframe."""
-    return _featurize_smiles_df(
-        shard,
-        self.featurizer,
-        field=self.smiles_field,
-        log_every_n=self.log_every_n)
+  def _featurize_shard(self,
+                       shard: pd.DataFrame) -> Tuple[np.ndarray, np.ndarray]:
+    """Featurizes a shard of an input dataframe.
+
+    Parameters
+    ----------
+    shard: pd.DataFrame
+      DataFrame that holds a shard of the input CSV file
+
+    Returns
+    -------
+    features: np.ndarray
+      Features computed from CSV file.
+    valid_inds: np.ndarray
+      Indices of rows in source CSV with valid data.
+    """
+    logger.info("About to featurize shard.")
+    features = [elt for elt in self.featurizer(shard[self.feature_field])]
+    valid_inds = np.array(
+        [1 if np.array(elt).size > 0 else 0 for elt in features], dtype=bool)
+    features = [
+        elt for (is_valid, elt) in zip(valid_inds, features) if is_valid
+    ]
+    return np.array(features), valid_inds
 
 
 class UserCSVLoader(CSVLoader):
   """
   Handles loading of CSV files with user-defined featurizers.
```

Comment:
Should that be "user-defined features" instead of "user-defined featurizers"?

Reply:
Ah good catch! Will fix